internal/internal_workflow.go

```diff
@@ -646,11 +661,26 @@ func getState(ctx Context) *coroutineState {
 	}
 	state := s.(*coroutineState)
 	if !state.dispatcher.IsExecuting() {
-		panic(panicIllegalAccessCoroutinueState)
+		panic(panicIllegalAccessCoroutineState)
 	}
 	return state
 }
 
+func isReadOnlyState(executionState ExecutionState) bool {
```

Comment:
```suggestion
func (e ExecutionState) isReadOnlyState() bool {
```

Just reads a bit clearer to me and doesn't pollute the package-level function set, but your choice. Or better yet just move this little logic inside of `assertNotInReadOnlyState` and not have a separate function that is one line and only used in one place.